lib/compress-stream.js

```diff
@@ -52,18 +60,25 @@ CompressStream.prototype._uncompressed = function (chunk) {
 
 CompressStream.prototype._transform = function (chunk, enc, callback) {
   var self = this
+  async function compressChunks() {
```

Comment:
Why async function?

Reply:
the API of transform is to call callback on the completion of transformation of the chunk

Reply:
But there is no await in this body, and `compressChunks()` is called without any handling. Errors will become unhandled rejections. Can you just make it
```ts
  function compressChunks() {
```